### PR TITLE
Added idle-js types

### PR DESCRIPTION
**Pull request by jpeirson**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/idle-js/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+declare namespace IdleJs {
+    /**
+     * Idle tracking options.
+     */
+    interface Options {
+        /**
+         * Idle time in ms. The default is 10000ms.
+         */
```
Comment (martin-badin) [inline types/idle-js/index.d.ts:51]: Please, could you add default values to comments for each option? Example

```suggestion
        /**
         * Idle time in ms.
         * @defaut 10000
         */
```

File: types/idle-js/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+     * @throws
+     */
+    constructor(options?: IdleJs.Options);
+
+    /**
+     * Starts tracking idle state.
+     *
+     * @returns this instance for call chaining
+     */
+    start(): IdleJs;
```
Comment (martin-badin) [inline types/idle-js/index.d.ts:20]: change `IdleJs` to `this`
```suggestion
    start(): this;
```

File: types/idle-js/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+    /**
+     * Stops all tracking.
+     *
+     * @returns this instance for call chaining
+     */
+    stop(): IdleJs;
```
Comment (martin-badin) [inline types/idle-js/index.d.ts:27]: change `IdleJs` to `this`
```suggestion
    stop(): this;
```

File: types/idle-js/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+    /**
+     * Reset visible and idle state.
+     *
+     * @param idle The new idle state, or undefined to use the existing state.
+     * @param visible The new visibility state, or undefined to use the existing state.
+     * @returns this instance for call chaining
+     */
+    reset(idle?: boolean, visible?: boolean): IdleJs;
```
Comment (martin-badin) [inline types/idle-js/index.d.ts:36]: change `IdleJs` to `this`

```suggestion
    reset(idle?: boolean, visible?: boolean): this;
```

**Review by martin-badin — CHANGES_REQUESTED** [review]

**Review by martin-badin — APPROVED** [review]
Thanks. Looks good to me

File: types/idle-js/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+         * Set it to true if you want `onIdle` to be raised each time the idle period elapses, not just on idle state changes;
+         * i.e.; set to true for `setInterval()` behavior, otherwise for `setTimeout()` behavior.
+         *
+         * @default false
+         */
+        recurIdleCall?: boolean;
+    }
+}
+
+export default IdleJs;
```
Comment (weswigham) [inline types/idle-js/index.d.ts:99]: [This should probably be](https://github.com/soixantecircuits/idle-js/blob/master/src/index.js#L135)
```suggestion
export = IdleJs;
```

**Review by weswigham — CHANGES_REQUESTED** [review]

**Review by martin-badin — APPROVED** [review]

**Review by weswigham — APPROVED** [review]

**Comment by jpeirson** [comment]
Ready to merge

**Pull request merged**